### PR TITLE
Creating a new E2E workflow on GHA

### LINKED ISSUE #754: [FeatureRequest] Python 3.9 CI pipeline
Create a CI pipeline for Python 3.9.

Officially launched: https://www.python.org/downloads/release/python-390/
More Information: https://docs.python.org/release/3.9.0/whatsnew/3.9.html

### PR DESCRIPTION
**Pull request by vrdmr**
## Description
Third workflow for the repo - Replacing the AzureDevops for GHA.

Fixes #754

---

### PR information

- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: .github/workflows/e2e_ci_workflow.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# This workflow will install Python dependencies and run end to end tests with single version of Python
```
Comment (vrdmr) [inline .github/workflows/e2e_ci_workflow.yml:1]: The new E2E workflow.

File: .github/workflows/e2e_ci_workflow.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+          dotnet-version: '2.2.207'
+      - name: Install dependencies and the worker
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple -U -e .[dev]
+          python -m pip install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple -U azure-functions --pre
+          python setup.py build
+          python setup.py webhost
+      - name: Running 3.6 Tests
+        if: matrix.python-version == 3.6
```
Comment (vrdmr) [inline .github/workflows/e2e_ci_workflow.yml:45]: GHA doesn't have a good way to have selective environment variables, like the way we had in AzDO. This was the second-best approach I could come up with, after multiple attempts. I didn't want to have multiple workflow files for the E2E.

File: .github/workflows/perf-testing-setup.yml [hunk 3]
```diff
@@ -2,7 +2,6 @@ name: Throughput testing workflow
 
 on:
   workflow_dispatch:
-  push:
```
Comment (vrdmr) [inline .github/workflows/perf-testing-setup.yml:5]: Removed throughput tests for each push. `workflow_dispatch` allows having a manual run on any branch.

File: .github/workflows/perf-testing-setup.yml [hunk 4]
```diff
@@ -24,10 +23,10 @@ jobs:
       uses: actions/setup-python@v2
       with:
         python-version: ${{ env.PYTHON_VERSION }}
-    - name: Set up Dotnet 3.1.x
+    - name: Set up Dotnet 2.x
       uses: actions/setup-dotnet@v1
       with:
-        dotnet-version: '3.1.x'
+        dotnet-version: '2.2.207'
```
Comment (vrdmr) [inline .github/workflows/perf-testing-setup.yml:29]: Version 3.1 doesn't build the host. This was failing.

File: .github/workflows/ut_ci_workflow.yml [hunk 5]
```diff
@@ -4,16 +4,21 @@
 name: CI Unit tests
 
 on:
+  schedule:
+    # Monday to Thursday 1 AM PDT build
+    # * is a special character in YAML so you have to quote this string
+    - cron: "0 8 * * 1,2,3,4"
```
Comment (vrdmr) [inline .github/workflows/ut_ci_workflow.yml:10]: Added a schedule similar to AzDO.

File: .github/workflows/ut_ci_workflow.yml [hunk 6]
```diff
@@ -4,16 +4,21 @@
...
   pull_request:
-    branches: [ dev, master, main ]
+    branches: [ dev, master, main, release/* ]
 
 jobs:
   build:
     name: "Python UT CI Run"
     runs-on: ubuntu-latest
     strategy:
+      fail-fast: true
```
Comment (vrdmr) [inline .github/workflows/ut_ci_workflow.yml:21]: Making the fail-fast explicit. The other tests have `fail-fast: false` to run them completely.

File: azure-pipelines.yml [hunk 7]
```diff
@@ -1,24 +1,11 @@
...
-- cron: "0 8 * * 1,2,3,4"
-  displayName: Monday to Thursday 1 AM build
-  branches:
-    include:
-      - dev
-      - master
-      - release/*
-    exclude:
-    - releases/ancient/*
-
```
Comment (vrdmr) [inline azure-pipelines.yml:21]: Removing the schedule now.

File: azure-pipelines.yml [hunk 8]
```diff
@@ -1,24 +1,11 @@
 name: $(Date:yyyyMMdd).$(Rev:r)
 
 trigger:
-- dev
-- master
```
Comment (vrdmr) [inline azure-pipelines.yml:5]: This pipeline is now useful only for release/*

File: azure-pipelines.yml [hunk 9]
```diff
@@ -64,21 +33,6 @@ jobs:
...
-      LINUXCOSMOSDBCONNECTIONSTRING: $(linuxCosmosDB)
-      LINUXEVENTHUBCONNECTIONSTRING: $(linuxEventHub)
-      LINUXSERVICEBUSCONNECTIONSTRING: $(linuxServiceBus)
-      LINUXEVENTGRIDTOPICURI: $(linuxEventGridTopicUri)
-      LINUXEVENTGRIDTOPICCONNECTIONKEY: $(linuxEventGridConnectionKey)
-    displayName: 'E2E Tests'
-  - task: PublishCodeCoverageResults@1
-    inputs:
-        codeCoverageTool: cobertura
-        summaryFileLocation: coverage.xml
```
Comment (vrdmr) [inline azure-pipelines.yml:81]: Removing the E2E stuff now.

File: tests/endtoend/test_blob_functions.py [hunk 10]
```diff
@@ -132,11 +128,10 @@ def test_blob_trigger(self):
 
     @testutils.retryable_test(3, 5)
     def test_blob_trigger_with_large_content(self):
-        datetime_iso = datetime.utcnow().isoformat()
-        data = datetime_iso * 1024 * 1024  # 26 MB
+        data = 'DummyDataDummyDataDummyData' * 1024 * 1024  # 27 MB
```
Comment (vrdmr) [inline tests/endtoend/test_blob_functions.py:131]: Python 3.9 becomes very slow and sometimes, previous runs have their files remaining in the system. I'll be moving the date-time logic to a guid-based logic. This is an intermediate step for now to remove reliance on date-times in the tests.

File: .github/workflows/perf-testing-setup.yml (hunk 4 above)
Comment (Hazhzeng) [reply .github/workflows/perf-testing-setup.yml:29]: Yeah, I think to enable 3.1 dotnet build, we also need to update the host link in setup.py https://github.com/Azure/azure-functions-python-worker/blob/dev/setup.py#L21.

But this host update should be placed in a separate PR, don't need to worry about it for now.

File: tests/endtoend/test_blob_functions.py (hunk 10 above)
Comment (Hazhzeng) [reply tests/endtoend/test_blob_functions.py:131]: Hmm, this is interesting, where does this slow comes from? It is when running this test_blob_io_large_bytes testcase? Or does it comes from datetime.utcnow()? I don't mind the intermediate step for changing it to static string for now, but this still introduce another task to move it back to guid eventually.

**Review by Hazhzeng — APPROVED** [review]

**Pull request merged**